eis_toolkit/raster_processing/unique_combinations.py

```diff
@@ -0,0 +1,68 @@
+from typing import List, Dict
+
+import itertools
+import numpy as np
+import rasterio
+
+from eis_toolkit.exceptions import InvalidParameterValueException
+
+
+def _unique_combinations(  # type: ignore[no-any-unimported]
+    list_of_rasters: List[rasterio.io.DatasetReader],
+    path_to_output_file: str
+) -> rasterio.io.DatasetReader:
+
+    bands = []
+    for raster in list_of_rasters:
+        bands.append(raster.read())
```

Comment:
I think here you should read the raster band by band and append each band separately, because now you are reading the whole data array with all of its bands straight to the bands list. So each element of the list will be 3D array, even if the raster has only one band.